python/ffsim/tenpy/gates/abstract_gates.py

```diff
@@ -0,0 +1,81 @@
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+
+import numpy as np
+import tenpy.linalg.np_conserved as npc
+from tenpy.algorithms.tebd import TEBDEngine
+from tenpy.linalg.charges import LegPipe
+from tenpy.networks.site import SpinHalfFermionSite
+
+# ignore lowercase argument and variable checks to maintain TeNPy naming conventions
+# ruff: noqa: N803, N806
+
+# define sites
+shfs = SpinHalfFermionSite(cons_N="N", cons_Sz="Sz")
+shfsc = LegPipe([shfs.leg, shfs.leg])
+
+
+def apply_single_site(eng: TEBDEngine, U1: np.ndarray, site: int) -> None:
+    r"""Apply a single-site gate to an MPS.
+
+    Args:
+        eng: The TEBD Engine.
+        U1: The single-site quantum gate.
+        site: The gate will be applied to `site` on the MPS.
+
+    Returns:
+        None
+    """
+
+    # apply single-site gate
+    U1_npc = npc.Array.from_ndarray(U1, [shfs.leg, shfs.leg.conj()], labels=["p", "p*"])
+    psi = eng.get_resume_data()["psi"]
+    psi.apply_local_op(site, U1_npc)
+
+
+def apply_two_site(
+    eng: TEBDEngine,
+    U2: np.ndarray,
+    sites: tuple[int, int],
+    *,
+    norm_tol: float = 1e-5,
+) -> None:
+    r"""Apply a two-site gate to an MPS.
+
+    Args:
+        eng: The TEBD Engine.
+        U2: The two-site quantum gate.
+        sites: The gate will be applied to adjacent sites `(site1, site2)` on the MPS.
+        norm_tol: The norm error above which we recanonicalize the MPS.
```

Comment:
Could you expand the explanation of this argument to make it more accessible to people who are not too familiar with the technical details of MPS?